### PR TITLE
More config options

### PR DESCRIPTION
**Pull request by narthur157**
Didn't realize this had already gotten merged, can we merge this in too?

Commit messages were bad, should have mentioned throwing errors for some things now.

### CODE REVIEW

File: ractive-partials.js [hunk 1]
```diff
@@ -6,18 +6,33 @@ import module from 'module';
 let findPartial = /{{>\s?([^\s]+)\s?}}/gi;
 
 export function load(modulePath, require, done) {
-  // TODO: Support relative paths from '.'
-  var config = module.config();
+  const defaultDelim = '$';
+  var config = module.config(),
```
Comment (Aintaer) [inline ractive-partials.js]: `const`, one per line.

File: ractive-partials.js [hunk 2]
```diff
@@ -6,18 +6,33 @@ import module from 'module';
...
+      extension = config.fileExtension || 'mustache',
+      invalidDelims = '@#^&*()+<>\/\\|=;~`%.,{}[]';
```
Comment (Aintaer) [inline ractive-partials.js]: Change to regexp with character class

File: ractive-partials.js [hunk 3]
```diff
@@ -6,18 +6,33 @@ import module from 'module';
...
 
   if (config.pathPrefix) {
     modulePath = `${config.pathPrefix}${modulePath}`;
   }
-  text.get(`${modulePath}.mustache`, (text) => {
+
+  // if config.pathDelimeter is invalid, reset to default
+  for (let letter of invalidDelims) {
```
Comment (Aintaer) [inline ractive-partials.js]: Terrible complexity. Regexp is specifically for this purpose. `invalidDelims.test(config.pathDelimiter)`

File: ractive-partials.js [hunk 4]
```diff
@@ -6,18 +6,33 @@ import module from 'module';
...
+    if (config.pathDelimeter.indexOf(letter) !== -1) {
+      console.warn(`Invalid config.pathDelimeter value: ${delim} replaced by ${defaultDelim}`);
+      // changing config.pathDelimeter prevents from getting this warning multiple times
+      config.pathDelimeter = defaultDelim;
```
Comment (Aintaer) [inline ractive-partials.js]: Do not modify the config object. It is read-only as far as this module is concerned.

File: ractive-partials.js [hunk 5]
```diff
@@ -6,18 +6,33 @@ import module from 'module';
...
+    }
+  }
+  var delim = config.pathDelimeter;
```
Comment (Aintaer) [inline ractive-partials.js]: This should be a `let` and moved above the test.

File: ractive-partials.js [hunk 6]
```diff
@@ -6,18 +6,33 @@ import module from 'module';
...
+  // prevent .mustache.mustache
+  modulePath.replace(`\.${extension}$`, '');
```
Comment (Aintaer) [inline ractive-partials.js]: This does not work. This is a string comparison/replacement, not a Regexp compile.

Should be
```js
const extensionCheck = new Regexp(`\.${extension}$`);
modulePath = extensionCheck.test(modulePath) ?
  modulePath :
  `${modulePath}.${extension}`;
```

File: bower.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "rapper",
-  "version": "0.2.3",
+  "name": "ractive-partials",
+  "version": "0.2.4",
   "main": "ractive-partials.js",
   "dependencies": {
     "ractive": "0.7.3",
```
Comment (Aintaer) [inline bower.json]: This is a hard version lock. Instead you should probably be doing `^0.7` or something.

File: dist/ractive-partials.js [hunk 8]
```diff
@@ -18,18 +18,25 @@ define(['exports', 'ractive', './text', 'module'], function (exports, _ractive,
...
+        extension = config.fileExtension || 'mustache',
+        invalidDelims = '@#^&*()+<>/\\|=;~`%.,{}[]';
 
     if (config.pathPrefix) {
       modulePath = '' + config.pathPrefix + '' + modulePath;
     }
-    _text2['default'].get('' + modulePath + '.mustache', function (text) {
+
+    // prevent .mustache.mustache
+    modulePath.replace('.' + extension + '$', '');
```
Comment (Aintaer) [inline dist/ractive-partials.js]: This. Does. Not. Work. See previous comment.

File: ractive-partials.js [hunk 9]
```diff
@@ -6,18 +6,25 @@ import module from 'module';
...
+      extension = config.fileExtension || 'mustache',
+      invalidDelims = '@#^&*()+<>\/\\|=;~`%.,{}[]';
 
   if (config.pathPrefix) {
     modulePath = `${config.pathPrefix}${modulePath}`;
   }
-  text.get(`${modulePath}.mustache`, (text) => {
+
+  // prevent .mustache.mustache
+  modulePath.replace(`\.${extension}$`, '');
```
Comment (Aintaer) [inline ractive-partials.js]: This. Does. Not. Work.

File: bower.json (hunk 7 above)
Comment (mrjoelkemp) [inline bower.json]: Please use `npm install --save my-sweet-module` when adding modules to the package.json file

Comment (mrjoelkemp) [inline bower.json]: Whoops, bower

File: ractive-partials.js (hunk 9 above)
Comment (mrjoelkemp) [inline ractive-partials.js]: `modulePath.replace` returns the replaced value. It does not mutate the string directly.

File: ractive-partials.js [hunk 10]
```diff
@@ -6,18 +6,24 @@ import module from 'module';
 let findPartial = /{{>\s?([^\s]+)\s?}}/gi;
 
 export function load(modulePath, require, done) {
-  // TODO: Support relative paths from '.'
-  var config = module.config();
+  const defaultDelim = '$';
```
Comment (Aintaer) [inline ractive-partials.js]: This is unused.

File: ractive-partials.js (hunk 9 above)
Comment (Aintaer) [inline ractive-partials.js]: Not only that, when `String#replace(string, string)` is the signature, that is a _literal_ match against the first string to replace with the second. Not a RegExp match, which you need it to be to enforce the end-of-string behavior.

**Comment by mrjoelkemp** [comment]
Would love tests for this library, if possible. Probably warrants a larger discussion.

**Comment by narthur157** [comment]
Could write some jasmine tests easily enough. I've just been using the samples I wrote to make sure things still work. I'm not sure how easy it is to test multiple config files in the same repo though

**Comment by Aintaer** [comment]
:+1:

**Comment by narthur157** [comment]
no way

**Comment by narthur157** [comment]
can you release this as 0.2.4 too?

**Pull request merged**